src/fsharp/service/ServiceLexing.fs

```diff
@@ -795,3 +795,629 @@ module Keywords =
+            | FSharpSyntaxTokenKind.UInt64
+            | FSharpSyntaxTokenKind.Int8
+            | FSharpSyntaxTokenKind.Int16
+            | FSharpSyntaxTokenKind.Int32
+            | FSharpSyntaxTokenKind.Int64
+            | FSharpSyntaxTokenKind.Ieee32
+            | FSharpSyntaxTokenKind.Ieee64
+            | FSharpSyntaxTokenKind.BigNumber -> true
+            | _ -> false
+
+        member this.IsCommentTrivia =
+            match this.Kind with
+            | FSharpSyntaxTokenKind.CommentTrivia
+            | FSharpSyntaxTokenKind.LineCommentTrivia -> true
+            | _ -> false
+
+    let lexWithErrorLogger (text: ISourceText) (filePath: string) conditionalCompilationDefines (flags: FSharpLexerFlags) supportsFeature errorLogger onToken pathMap (ct: CancellationToken) =
+        let canSkipTrivia = (flags &&& FSharpLexerFlags.SkipTrivia) = FSharpLexerFlags.SkipTrivia
+        let isLightSyntaxOn = (flags &&& FSharpLexerFlags.LightSyntaxOn) = FSharpLexerFlags.LightSyntaxOn
+        let isCompiling = (flags &&& FSharpLexerFlags.Compiling) = FSharpLexerFlags.Compiling
+        let isCompilingFSharpCore = (flags &&& FSharpLexerFlags.CompilingFSharpCore) = FSharpLexerFlags.CompilingFSharpCore
+        let canUseLexFilter = (flags &&& FSharpLexerFlags.UseLexFilter) = FSharpLexerFlags.UseLexFilter
+
+        let lexbuf = UnicodeLexing.SourceTextAsLexbuf(supportsFeature, text)
+        let lightSyntaxStatus = LightSyntaxStatus(isLightSyntaxOn, true) 
+        let lexargs = mkLexargs (filePath, conditionalCompilationDefines, lightSyntaxStatus, Lexhelp.LexResourceManager(0), [], errorLogger, pathMap)
+        let lexargs = { lexargs with applyLineDirectives = isCompiling }
+
+        let getNextToken =
+            let lexer = Lexer.token lexargs canSkipTrivia
+
+            if canUseLexFilter then
+                LexFilter.LexFilter(lexargs.lightSyntaxStatus, isCompilingFSharpCore, lexer, lexbuf).Lexer
+            else
+                lexer
+
+        use _unwindBP = PushThreadBuildPhaseUntilUnwind BuildPhase.Parse
+        use _unwindEL = PushErrorLoggerPhaseUntilUnwind (fun _ -> DiscardErrorsLogger)
+
+        usingLexbufForParsing (lexbuf, filePath) (fun lexbuf -> 
+            while not lexbuf.IsPastEndOfStream do
+                ct.ThrowIfCancellationRequested ()
+                onToken (getNextToken lexbuf) lexbuf.LexemeRange)
+
+    let lex text filePath conditionalCompilationDefines flags supportsFeature lexCallback pathMap ct =
+        let errorLogger = CompilationErrorLogger("Lexer", ErrorLogger.FSharpErrorSeverityOptions.Default)
+        lexWithErrorLogger text filePath conditionalCompilationDefines flags supportsFeature errorLogger lexCallback pathMap ct
+
+    [<AbstractClass;Sealed>]
+    type FSharpLexer =
+
+        static member Lex(text: ISourceText, tokenCallback, ?langVersion, ?filePath, ?conditionalCompilationDefines, ?flags, ?pathMap, ?ct) =
+            let langVersion = defaultArg langVersion "latestmajor"
```

Comment:
Just to eliminate more magic strings, is there a constant defined in `LanguageVersion.fsi` that can be used here?

Reply:
We don't, but we can add them.

Reply:
So, I think we should wait to add them until VS has the F# compiler with the `Literal` attribute fix for signature files.

I want to add these strings but I can't make them literal due to the literal bug for signature files. It is a pain to not have them as literals. When 16.5 is released we can add them in.